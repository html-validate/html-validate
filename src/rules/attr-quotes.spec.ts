import HtmlValidate from '../htmlvalidate';

describe('rule attr-quotes', function(){

	const expect = require('chai').expect;

	let htmlvalidate: HtmlValidate;

	describe('with double-quote option', function(){

		before(function(){
			htmlvalidate = new HtmlValidate({
				rules: {'attr-quotes': ['error', {style: "double"}]},
			});
		});

		it('should not report when attributes use double quotes', function(){
			const report = htmlvalidate.validateString('<div foo="bar"></div>');
			expect(report).to.be.valid;
		});

		it('should report error when attributes use single quotes', function(){
			const report = htmlvalidate.validateString('<div foo=\'bar\'></div>');
			expect(report).to.be.invalid;
			expect(report).to.have.error('attr-quotes', 'Attribute "foo" used \' instead of expected "');
		});

	});

	describe('with single-quote option', function(){

		before(function(){
			htmlvalidate = new HtmlValidate({
				rules: {'attr-quotes': ['error', {style: "single"}]},
			});
		});

		it('should report error when attributes use double quotes', function(){
			const report = htmlvalidate.validateString('<div foo="bar"></div>');
			expect(report).to.be.invalid;
			expect(report).to.have.error('attr-quotes', 'Attribute "foo" used " instead of expected \'');
		});

		it('should not report when attributes use single quotes', function(){
			const report = htmlvalidate.validateString('<div foo=\'bar\'></div>');
			expect(report).to.be.valid;
		});

	});

	describe('with unquoted allowed', function(){

		before(function(){
			htmlvalidate = new HtmlValidate({
				rules: {'attr-quotes': ['error', {style: "double", unquoted: true}]},
			});
		});

		it('should not report when attributes is using quotes', function(){
			const report = htmlvalidate.validateString('<div foo="bar"></div>');
			expect(report).to.be.valid;
		});

		it('should not report error when attribute value is unquoted', function(){
			const report = htmlvalidate.validateString('<div foo=5></div>');
			expect(report).to.be.valid;
		});

	});

	describe('with unquoted disabled', function(){

		before(function(){
			htmlvalidate = new HtmlValidate({
				rules: {'attr-quotes': ['error', {style: "double", unquoted: false}]},
			});
		});

		it('should not report when attributes is using quotes', function(){
			const report = htmlvalidate.validateString('<div foo="bar"></div>');
			expect(report).to.be.valid;
		});

		it('should report error when attribute value is unquoted', function(){
			const report = htmlvalidate.validateString('<div foo=5></div>');
			expect(report).to.be.invalid;
			expect(report).to.have.error('attr-quotes', 'Attribute "foo" using unquoted value');
		});

	});

});
