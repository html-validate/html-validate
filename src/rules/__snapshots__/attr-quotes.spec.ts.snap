// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rule attr-quotes should contain documentation style "any" and unquoted "false" 1`] = `
"Attribute \`foo\` must not be unquoted.

Under the current configuration attributes must be:

- quoted with single quotes \`'\` or
- quoted with double quotes \`"\`
"
`;

exports[`rule attr-quotes should contain documentation style "any" and unquoted "true" 1`] = `
"Attribute \`foo\` must not be unquoted.

Under the current configuration attributes must be:

- quoted with single quotes \`'\` or
- quoted with double quotes \`"\` or
- unquoted (if applicable)
"
`;

exports[`rule attr-quotes should contain documentation style "auto" and unquoted "false" 1`] = `
"Attribute \`foo\` must not be unquoted.

Under the current configuration attributes must be:

- quoted with double quotes \`"\` unless the value contains double quotes in which case single quotes \`'\` should be used instead
"
`;

exports[`rule attr-quotes should contain documentation style "auto" and unquoted "true" 1`] = `
"Attribute \`foo\` must not be unquoted.

Under the current configuration attributes must be:

- quoted with double quotes \`"\` unless the value contains double quotes in which case single quotes \`'\` should be used instead or
- unquoted (if applicable)
"
`;

exports[`rule attr-quotes should contain documentation style "double" and unquoted "false" 1`] = `
"Attribute \`foo\` must not be unquoted.

Under the current configuration attributes must be:

- quoted with double quotes \`"\`
"
`;

exports[`rule attr-quotes should contain documentation style "double" and unquoted "true" 1`] = `
"Attribute \`foo\` must not be unquoted.

Under the current configuration attributes must be:

- quoted with double quotes \`"\` or
- unquoted (if applicable)
"
`;

exports[`rule attr-quotes should contain documentation style "single" and unquoted "false" 1`] = `
"Attribute \`foo\` must not be unquoted.

Under the current configuration attributes must be:

- quoted with single quotes \`'\`
"
`;

exports[`rule attr-quotes should contain documentation style "single" and unquoted "true" 1`] = `
"Attribute \`foo\` must not be unquoted.

Under the current configuration attributes must be:

- quoted with single quotes \`'\` or
- unquoted (if applicable)
"
`;

exports[`rule attr-quotes should contain documentation with style context 1`] = `
"Attribute \`foo\` must use \`"\` instead of \`'\`.

Under the current configuration attributes must be:

- quoted with double quotes \`"\` unless the value contains double quotes in which case single quotes \`'\` should be used instead
"
`;

exports[`rule attr-quotes should contain documentation with unquoted context 1`] = `
"Attribute \`foo\` must not be unquoted.

Under the current configuration attributes must be:

- quoted with double quotes \`"\` unless the value contains double quotes in which case single quotes \`'\` should be used instead
"
`;
